added a default option to seed

RugnirViking opened the pull request:
added a default option to the seed function which causes the seed to be the current system time in ns. Allows for simple pseudorandom seed choice. It was something I needed when I was using the library, so made a pr as a possible nice-to-have option, only adds time module from python standard library as an import.

RugnirViking commented:
I notice it could be a merge conflict with the other open PR but should be a simple enough fix, just set the parameter for that function to both have typing and be default

lmas commented:
Thank you, really appreciate it and especially that you took time to check for conflicts with other PRs too!

I need to think about this one too though. I think this would break the principle of least surprises by having the seed change on every run, might cause problems for someone else just trying to get things running.

_Having said that, I realise that the default seed the library is already using is hidden away in the internals, which sure would cause surprises!_

RugnirViking commented:
It should only change the seed when the user calls seed(), so hopefully not too unexpected. Still, do have a think about it. Perhaps it could be a separate function instead.

lmas commented:
On 18 February 2022 16:30:00 UTC, Samuel Howell ***@***.***> wrote:
>Perhaps it could be a separate function instead. 


Yeah I like this way better; a separate "randomize seed" func

lmas commented:
I finally got around to run the tests, fixed a bug for ya and cleaned up. 

Before I merge, are you ok with my commit?

RugnirViking commented:
> I finally got around to run the tests, fixed a bug for ya and cleaned up.
> 
> Before I merge, are you ok with my commit?

yep, looks mostly good. Only one extremely minor thing, the comment in your commit says "it uses the system (in ns)" and it should probably read "it uses the system **time** (in ns)"

lmas commented:
Good eye, thanks! Pleasure working with you.
